node/node.go

```diff
@@ -374,76 +337,71 @@ func (node *Node) Start() error {
 			return err
 		}
 		topicNamed := allTopics[i].Name
+
+		if nodeconfig.GroupID(topicNamed) == nodeconfig.NewGroupIDByShardID(
+			nodeconfig.ShardID(node.Consensus.ShardID),
+		) {
+
+			pubsub.RegisterTopicValidator(
+				topicNamed,
+				func(ctx context.Context, peer libp2p_peer.ID, msg *libp2p_pubsub.Message) bool {
+					sig2 := msg.GetData()
+					if len(sig2) < p2pMsgPrefixSize {
+						return true
+					}
+
+					cnsMsg := sig2[p2pMsgPrefixSize:][proto.MessageCategoryBytes:]
+
+					var (
+						m         msg_pb.Message
+						senderKey *bls.PublicKey
+					)
+
+					if err := protobuf.Unmarshal(cnsMsg, &m); err != nil {
+						errChan <- err
+						return false
+					}
+
+					if c := m.GetConsensus(); c.GetSenderPubkey() != nil {
```

Comment:
Consensus can be nil. Also what about view change msg validation?

Reply:
Iirc view change is consensus as well , so this should cover it

Reply:
Ah ya, forgot to come back to && part

Reply:
But ya, admittedly , not sure for view change, can come back to in next PR, after have legit way to maybe spam attack that ?

Reply:
this check also need to happen for ViewChange right? as you removed the `VerifyMessageSig` check in `viewChangeSanityCheck`

Reply:
Sure - but viewchange is either already a consensus message or I will handle there - viewchange/consensus should not be doing those checks - they are huge waste of compute of the consensus thread

Reply:
yes, need `c := m.GetViewchange()` also.

Reply:
thank you!